Update index.md (#1547)
shopware/docs: resources/guidelines/testing/store/quality-guidelines-plugins/index.md

If an extension with the same function exists and it does not fit into one of our differentiator clusters, it can be rejected as it doesn't provide any added value.
If you would like more information, please write an email to [qa@shopware.com](mailto:qa@shopware.com).

> Link: [Differentiator cluster for Shopware extensions](https://developer.shopware.com/docs/resources/guidelines/testing/Differentiator-Clusters.html)
> Link: [Documentation for Extension Partner](https://docs.shopware.com/en/account-en/extension-partner/extensions?category=account-en/extension-partner#how-can-i-request-a-preview)

Reviewer: ```suggestion
[Differentiator cluster for Shopware extensions](https://developer.shopware.com/docs/resources/guidelines/testing/Differentiator-Clusters.html)
[Documentation for Extension Partner](https://docs.shopware.com/en/account-en/extension-partner/extensions?category=account-en/extension-partner#how-can-i-request-a-preview)
```

	You can embed a maximum of 2 YouTube videos in your app description.

::: info

> You can no longer advertise your Shopware certificates within the app description, in your app images, or in your manufacturer profile. The manufacturer/partner certificates are dynamically loaded at the end of each app description and published by us.

Reviewer 2: ```suggestion
You can no longer advertise your Shopware certificates within the app description, in your app images, or in your manufacturer profile.
The manufacturer/partner certificates are dynamically loaded at the end of each app description and published by us.
```

With `executeComposerCommands() === true` in the plugin base class, we provide a dynamic installation of the composer dependencies by default, so they don't have to be included.
Everything that is delivered in code should be traceable either directly or via `composer.json`.

> Link: [Developer documentation article to add private dependency](../../../../../guides/plugins/plugins/plugin-fundamentals/using-composer-dependencies)

Reviewer: ```suggestion
[Developer documentation article to add private dependency](../../../../../guides/plugins/plugins/plugin-fundamentals/using-composer-dependencies)
```

Please build your `main.js` as described in our documentation and create the minified code as described in our developer documentation. 

> Link: [Loading the JS files](https://developer.shopware.com/docs/guides/plugins/plugins/administration/add-custom-field.html#loading-the-js-files)
> Link: [Injecting into the Administration](https://developer.shopware.com/docs/guides/plugins/plugins/administration/add-custom-field.html#injecting-into-the-administration)

Reviewer: ```suggestion
[Loading the JS files](https://developer.shopware.com/docs/guides/plugins/plugins/administration/add-custom-field.html#loading-the-js-files)
[Injecting into the Administration](https://developer.shopware.com/docs/guides/plugins/plugins/administration/add-custom-field.html#injecting-into-the-administration)
```

There should not be any drastic change in performance, accessibility values, or any new errors when activating the app.

* **Testing tool** for A/B-Testing:

>   * Link: [Google Lighthouse](https://developers.google.com/web/tools/lighthouse)

Reviewer: ```suggestion
  * [Google Lighthouse](https://developers.google.com/web/tools/lighthouse)
```

### schema.org/Rich Snippets A/B-Testing:

Do an A/B-Test with *Scheme.org's Structured Data Testing Tool* and *Google Rich Result Tester* to check the homepage, categories, and various product detail pages (incl. available products, unavailable products, products with no review, single review, many reviews with various ratings, out-of-stock products, products to be released in the future or any other kind of product configuration and products including ean, mpn, width, length, height, weight).
Also, could you check for duplicate entries as well as any new bugs?

* **Testing tool** for A/B-Testing:

>   * Link: [Schema Markup Validator of schema.org](https://validator.schema.org/)
>   * Link: [Google Rich Result Tester](https://search.google.com/test/rich-results)

Reviewer: ```suggestion
  * [Schema Markup Validator of schema.org](https://validator.schema.org/)
  * [Google Rich Result Tester](https://search.google.com/test/rich-results)
```

We expect every cookie set from the store URL to be optional and not technically required for running shopware.
Therefore, the cookies had to be [registered in our Cookie Consent Manager](/docs/guides/plugins/plugins/storefront/add-cookie-to-manager).

> We differentiate between "Technically required","Marketing" and "Comfort features".

Reviewer 2: ```suggestion
We differentiate between "Technically required", "Marketing" and "Comfort features".
```

If the API data is incorrect, an entry must appear in the event log file in the Shopware folder `/var/log/` respectively in the database.

* **Example for implementing an API Test Button into the System Config form:

>  * Link: [Github](https://github.com/shyim/ShyimApiTest)

Reviewer: ```suggestion
 * [Github](https://github.com/shyim/ShyimApiTest)
```

Our most current code review configurations when uploading apps via the Shopware Account can be found on GitHub.

> * Link: [Code reviews for Shopware 6 on GitHub](https://github.com/shopwareLabs/store-plugin-codereview)

Reviewer: ```suggestion
* [Code reviews for Shopware 6 on GitHub](https://github.com/shopwareLabs/store-plugin-codereview)
```

### Sonarcube Rules status Blocker

The following statements will be blocked as of 1st Oct. 2022:  
-die; exit; var_dump

> * Link: [Refer to the list of the already existing blockers](https://s3.eu-central-1.amazonaws.com/wiki-assets.shopware.com/1657519735/blocker.txt).

Reviewer: ```suggestion
* [Refer to the list of the already existing blockers](https://s3.eu-central-1.amazonaws.com/wiki-assets.shopware.com/1657519735/blocker.txt).
```

### Automated code tests with Cypress

There are Cypress tests for Shopware 6 on GitHub.
The project is driven by the *Friends of Shopware* group. You can contribute at any time:

> * Link: [Developer Documentation Cypress Tests for Shopware 6](/docs/guides/plugins/plugins/testing/end-to-end-testing)
> * Link: [Cypress Tests for Shopware 6](https://github.com/shopware/shopware/tree/trunk/src/Administration/Resources)

Reviewer: ```suggestion
* [Developer Documentation Cypress Tests for Shopware 6](/docs/guides/plugins/plugins/testing/end-to-end-testing)
* [Cypress Tests for Shopware 6](https://github.com/shopware/shopware/tree/trunk/src/Administration/Resources)
```

### Helpful tools for app developers

> * Link: [FroshPluginUploader](https://github.com/FriendsOfShopware/FroshPluginUploader): Tool for validating and uploading new SW6 app releases to the Community Store (GitHub Project from "Friends of Shopware")]
> * Link: [Shopware CLI tools](https://github.com/shopwareLabs/sw-cli-tools): When you think about performance, these are various useful console helpers for generating data.

Reviewer: ```suggestion
* [FroshPluginUploader](https://github.com/FriendsOfShopware/FroshPluginUploader): Tool for validating and uploading new SW6 app releases to the Community Store (GitHub Project from "Friends of Shopware")]
* [Shopware CLI tools](https://github.com/shopwareLabs/sw-cli-tools): When you think about performance, these are various useful console helpers for generating data.
```

**Cause:** Error in composer.json  

One possible cause is that the technical app name from the Community Store or Account does not match the technical name entered in composer.json, or the app is incorrectly zipped.

> The technical app name must be stored in the first part of composer.json, located at composer.json > extra > shopware-plugin-class.

Reviewer 2: it does not really matter where this located in the composer.json :thinking:

Could you take a look at the bootstrap class? Most of the errors are caused by the wrong technical name.
For example, "Swag\\MyPlugin\\SwagMyPluginSW6" instead of "Swag\\MyPlugin\\SwagMyPlugin".

> Link: [Exampl eof a valid composer.json](https://github.com/FriendsOfShopware/FroshPlatformPerformance/blob/master/composer.json#L20).  

Reviewer 2: ```suggestion
Link: [Example of a valid composer.json](https://github.com/FriendsOfShopware/FroshPlatformPerformance/blob/master/composer.json#L20).  
```

**Cause:** Error in composer.json  

One possible cause is that the technical app name from the Community Store or Account does not match the technical name entered in composer.json, or the app is incorrectly zipped.

> The technical app name must be stored in the first part of composer.json, located at composer.json > extra > shopware-plugin-class.

Reviewer: ```suggestion
The technical app name must be stored in the composer.json, located at `composer.json` > extra > `shopware-plugin-class`.
```

Could you take a look at the bootstrap class? Most of the errors are caused by the wrong technical name.
For example, "Swag\\MyPlugin\\SwagMyPluginSW6" instead of "Swag\\MyPlugin\\SwagMyPlugin".

> Link: [Example of a valid composer.json](https://github.com/FriendsOfShopware/FroshPlatformPerformance/blob/master/composer.json#L20).  

Reviewer: ```suggestion
[Example of a valid composer.json](https://github.com/FriendsOfShopware/FroshPlatformPerformance/blob/master/composer.json#L20).  
```

Here are some examples of not allowed folders and files:
* /__MACOSX

> *./tests

Reviewer 2: ```suggestion
* /tests
```

this was just changed by you :thinking:
